src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisResultService.java

```diff
@@ -56,22 +60,36 @@ public SecurityAnalysisResultService(SecurityAnalysisResultRepository securityAn
     }
 
     @Transactional(readOnly = true)
-    public PreContingencyResult findNResult(UUID resultUuid) {
+    public List<PreContingencyLimitViolationResultDTO> findNResult(UUID resultUuid, String stringFilters, Sort sort) {
         Optional<SecurityAnalysisResultEntity> securityAnalysisResult = securityAnalysisResultRepository.findById(resultUuid);
         if (securityAnalysisResult.isEmpty()) {
             return null;
         }
-        List<PreContingencyLimitViolationEntity> preContingencyLimitViolationEntities = preContingencyLimitViolationRepository.findByResultId(resultUuid);
 
-        List<LimitViolation> preContingencyLimitViolations = preContingencyLimitViolationEntities.stream()
-            .map(AbstractLimitViolationEntity::toLimitViolation)
-            .toList();
+        Specification<PreContingencyLimitViolationEntity> specification = preContingencyLimitViolationRepository.getParentsSpecifications(resultUuid, fromStringFiltersToDTO(stringFilters));
+        Sort newSort = createSort(sort);
+        List<PreContingencyLimitViolationEntity> preContingencyLimitViolation = preContingencyLimitViolationRepository.findAll(specification, newSort);
 
-        return new PreContingencyResult(
-            LoadFlowResult.ComponentResult.Status.valueOf(securityAnalysisResult.get().getPreContingencyStatus()),
-            new LimitViolationsResult(preContingencyLimitViolations),
-            new NetworkResult(Collections.emptyList(), Collections.emptyList(), Collections.emptyList())
-        );
+        return preContingencyLimitViolation.stream()
+                .map(PreContingencyLimitViolationResultDTO::toDto)
+                .toList();
+    }
+
+    private Sort createSort(Sort sort) {
+        List<Sort.Order> newOrders = new ArrayList<>();
+        List<String> columnNames = ResourceFilterDTO.getAllColumnNames();
+        sort.forEach(order -> {
+            boolean columnExist = columnNames.stream().anyMatch(columnName -> columnName.equals(order.getProperty()));
```

Comment:
we are allowing the user to sort by "contingencyId" whereas it doesn't exist for this result